lib/jerboa/client.ex

```diff
@@ -29,41 +29,70 @@ defmodule Jerboa.Client do
   @type t :: pid
   @type port_no :: :inet.port_number
   @type ip :: :inet.ip_address
+  @type address :: {ip, port_no}
+  @type start_opts :: [start_opt]
+  @type start_opt :: {:server, address}
+                   | {:username, String.t}
+                   | {:secret, String.t}
+  @type error :: :bad_response
+               | :no_allocation
+               | Jerboa.Format.Body.Attribute.ErrorCode.name
 
   alias Jerboa.Client
 
   @doc """
   Starts STUN client process
 
-      iex> Jerboa.Client.start server: %{address: {192, 168, 1, 20}, port: 3478}
+      iex> opts = [server: {{192, 168, 1, 20}, 3478}, username: "user", secret: "abcd"]
+      iex> Jerboa.Client.start(opts)
       {:ok, #PID<...>}
 
   ### Options
 
-  * `server` - required - a map with `:address` and `:port` keys
+  * `:server` - required - a tuple with server's address and port
+  * `:username` - required - username used for authentication
+  * `:secret` - required - secret used for authentication
   """
   @spec start(options :: Keyword.t) :: Supervisor.on_start_child
-  def start(x) do
-    Supervisor.start_child(Client.Supervisor, [server(x)])
+  def start(opts) do
+    Supervisor.start_child(Client.Supervisor, [opts])
   end
 
   @doc """
   Sends Binding request to a server
 
-  Returns reflexive address and port on successful response. Otherwise the
-  client process crashes.
+  Returns reflexive address and port on successful response. Returns
+  error tuple if response from the server is invalid. Client process
+  crashes if response is not a valid STUN message.
   """
-  @spec bind(t) :: {ip, port_no} | no_return
+  @spec bind(t) :: {:ok, address} | {:error, :bad_response} | no_return
   def bind(client) do
-    GenServer.call(client, :bind)
+    GenServer.call(client, :bind, 2 * timeout())
```

Comment:
To be honest, I don't know why it it `2*`. What do you recommend, leave `timeout()` or go with the default GenServer timeout? If we go with the default, it doesn't make much sense to configure timeout at all.